### PR TITLE
feat(plasma-b2c): handle text field size

### PR DESCRIPTION
**Pull request by Yakutoc**
Добавлены размеры и связанные с ними отступы.

Так же поправлено согласно макетам типографика.    

:baby_chick: Download canary assets:

[colors--canary.210.3385095837.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.210.3385095837.xml)  
[color_sbermarket_ios-swift--canary.210.3385095837.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.210.3385095837.swift)  
[color_sbermarket_kotlin--canary.210.3385095837.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.210.3385095837.kt)  
[color_sbermarket_react-native--canary.210.3385095837.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.210.3385095837.ts)  
[color_sberprime_ios-swift--canary.210.3385095837.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.210.3385095837.swift)  
[color_sberprime_kotlin--canary.210.3385095837.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.210.3385095837.kt)  
[color_sberprime_react-native--canary.210.3385095837.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.210.3385095837.ts)  
[PlasmaTokensColor--canary.210.3385095837.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.210.3385095837.swift)  
[typo_mage_kotlin--canary.210.3385095837.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.210.3385095837.kt)  
[typo_mage_react-native--canary.210.3385095837.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.210.3385095837.ts)  
[typo_plasma_kotlin--canary.210.3385095837.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.210.3385095837.kt)  
[typo_plasma_react-native--canary.210.3385095837.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.210.3385095837.ts)  
[typo_ruler_kotlin--canary.210.3385095837.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.210.3385095837.kt)  
[typo_ruler_react-native--canary.210.3385095837.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.210.3385095837.ts)  
[typo_sage_kotlin--canary.210.3385095837.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.210.3385095837.kt)  
[typo_sage_react-native--canary.210.3385095837.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.210.3385095837.ts)  
[typo_sbermarket_kotlin--canary.210.3385095837.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.210.3385095837.kt)  
[typo_sbermarket_react-native--canary.210.3385095837.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.210.3385095837.ts)  
[typo_soulmate_kotlin--canary.210.3385095837.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.210.3385095837.kt)  
[typo_soulmate_react-native--canary.210.3385095837.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.210.3385095837.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.111.0-canary.210.3385095837.0
  # or 
  yarn add @salutejs/plasma-b2c@1.111.0-canary.210.3385095837.0
  ```
</details>

### CODE REVIEW

File: packages/plasma-b2c/src/components/TextField/Content.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import styled from 'styled-components';
+import { FieldContent } from '@salutejs/plasma-core';
+
+export const Content = styled(FieldContent)`
+    ${({ pos }) => (pos === 'left' ? 'left: 0' : 'right: 0')};
```
Comment (Yakutoc) [inline packages/plasma-b2c/src/components/TextField/Content.tsx:5]: Тут объединю логику, что бы 2 раз не делать практически одно и тоже.

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-210/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-210/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-210/
  ui storybook: http://plasma.sberdevices.ru/pr-210/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-210/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-210/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-210/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-210/
  ui storybook: http://plasma.sberdevices.ru/pr-210/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-210/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-210/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-210/temple-storybook/

File: packages/plasma-b2c/src/components/TextField/Content.tsx (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-b2c/src/components/TextField/Content.tsx:5]: в принципе тут можно две проверки в одну объединить. Иначе получается, что на 5 строке ты проверяешь на pos === left, и ту же проверку на 7

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-210/

File: packages/plasma-b2c/src/components/TextField/Placeholder.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import styled from 'styled-components';
+import { FieldPlaceholder } from '@salutejs/plasma-core';
+
+export const Placeholder = styled(FieldPlaceholder)<{ visible?: boolean }>`
+    --display: ${({ visible }) => (visible ? 'inline-flex' : 'none')};
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextField/Placeholder.tsx:5]: а почему тут нужна css переменная?

File: packages/plasma-b2c/src/components/TextField/TextField.stories.tsx [hunk 3]
```diff
@@ -85,6 +85,7 @@ Default.args = {
     status: '' as 'success',
     disabled: false,
     readOnly: false,
+    size: 'm',
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextField/TextField.stories.tsx:88]: у нас по дефолту он не выставляется?

File: packages/plasma-core/src/components/TextField/TextField.component-test.tsx [hunk 4]
```diff
@@ -32,6 +32,84 @@ describe('plasma-core: TextField', () => {
...
+        function TextFieldStub(props) {
+            return (
+                <TextField
+                    {...props}
+                    caption="Label"
+                    value="Input value"
+                    helperText="Helper text"
+                    placeholder="Placeholder"
+                    contentLeft={<IconSleep color="inherit" size="s" />}
+                    contentRight={<IconEye color="inherit" size="s" />}
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/TextField/TextField.component-test.tsx:65]: А иконки опять таки у нас могут быть только size=s , не больше?

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-210/
  ui storybook: http://plasma.sberdevices.ru/pr-210/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-210/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-210/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-210/temple-storybook/

File: packages/plasma-b2c/src/components/TextField/TextField.tsx [hunk 5]
```diff
@@ -1,52 +1,24 @@
...
-    }
-`;
-const StyledCaption = styled.span`
-    ${bodySBold}
-
-    display: flex;
-    margin-bottom: 0.75rem;
-    color: ${primary};
-`;
+    size?: 'l' | 'm' | 's' | 'xs';
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextField/TextField.tsx:20]: напиши jsdoc
что-то такое:

```
 /**
  * Размер поля
  * /
```

File: packages/plasma-b2c/src/components/TextField/TextField.stories.tsx (hunk 3 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextField/TextField.stories.tsx:88]: > у нас по дефолту он не выставляется?

Не понял вопроса.

File: packages/plasma-core/src/components/TextField/TextField.component-test.tsx (hunk 4 above)
Comment (Yakutoc) [reply packages/plasma-core/src/components/TextField/TextField.component-test.tsx:65]: @Yeti-or Здесь приняли совместное решение, что у нас на стороне компонента будет идти ограничение размера иконки при размере XS иконка будет 16х16. 

В остальных случаях - 24х24.

File: packages/plasma-b2c/src/components/TextField/Placeholder.tsx (hunk 2 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextField/Placeholder.tsx:5]: Пошел по такому пути, потому что когда делал так 

```
input:not(:placeholder-shown) ~ &  {
     display: ${({ visible }) => (visible ? 'inline-flex' : 'none')};
}
```

Поведение было неправильным.

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-b2c/src/components/TextField/properties.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
+import { css } from 'styled-components';
+
+export const properties = {
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/TextField/properties.ts:3]: я бы переименовал этот файл по аналогии с другими компонентами, где в отдельных мапах лежат токены. Например в TextField.props.ts

File: packages/plasma-b2c/src/components/TextField/Field.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import styled, { SimpleInterpolation } from 'styled-components';
+import { FieldRoot } from '@salutejs/plasma-core';
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/TextField/Field.tsx:2]: а Field, который уже лежит в plasma-b2c не подойдёт? Просто в таком случае получается дублирование части кода 🤔

File: packages/plasma-b2c/src/components/TextField/properties.ts (hunk 6 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextField/properties.ts:3]: Хорошо

File: packages/plasma-b2c/src/components/TextField/Input.tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import styled from 'styled-components';
+import { Input as InputCore } from '@salutejs/plasma-core';
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/TextField/Input.tsx:2]: тут тоже поправь пожалуйста на импорт из "@salutejs/plasma-web". Идея в том, что библиотека plasma-b2c (на текущий момент) расширяет или переиспользует компоненты, которые есть в plasma-web

File: packages/plasma-b2c/src/components/TextField/Field.tsx (hunk 7 above)
Comment (neretin-trike) [reply packages/plasma-b2c/src/components/TextField/Field.tsx:2]: договорились, что сделаем этот переезд в отдельной задаче

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-210/
  ui storybook: http://plasma.sberdevices.ru/pr-210/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-210/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-210/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-210/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-210/
  ui storybook: http://plasma.sberdevices.ru/pr-210/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-210/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-210/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-210/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-210/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-210/
  ui storybook: http://plasma.sberdevices.ru/pr-210/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-210/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-210/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-210/temple-storybook/

**Pull request merged**